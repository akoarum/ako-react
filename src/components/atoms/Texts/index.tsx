import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'

type Props = {
  className?: string
  caution?: boolean
  small?: boolean
}

export const StyledTexts = styled.div`
  white-space: pre-line;
  word-wrap: break-word;

  ${style.mediaQuery.pc} {
    font-size: ${style.font.sizePc};
    line-height: ${style.font.lineHeightPc};
  }

  ${style.mediaQuery.sp} {
    font-size: ${style.font.sizeSp};
    line-height: ${style.font.lineHeightSp};
  }

  &.-caution {
    color: ${style.color.caution};
  }

  &.-small {
    ${style.mediaQuery.pc} {
      font-size: 1.4rem;
      line-height: 1.6;
    }
    ${style.mediaQuery.sp} {
      font-size: 1.3rem;
      line-height: 1.5;
    }
  }
`

export const getClassNames = (
  className?: string,
  caution?: boolean,
  small?: boolean
): string[] => {
  const classNames = []

  if (className) {
    classNames.push(className)
  }

  if (caution) {
    classNames.push('-caution')
  }

  if (small) {
    classNames.push('-small')
  }

  return classNames
}

export const Texts: React.FC<Props> = ({
  className,
  caution,
  small,
  children
}) => {
  const classNames: string[] = getClassNames(className, caution, small)

  return <StyledTexts className={classNames.join(' ')}>{children}</StyledTexts>
}

export default Texts
