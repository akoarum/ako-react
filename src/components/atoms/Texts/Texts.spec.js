import React from 'react'
import renderer from 'react-test-renderer'
import Texts, { getClassNames } from './index'

describe('Texts', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderer.create(
      <Texts />
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('getClassNames: propsをもとにクラスを付与して返す', () => {
    it('特に何も無い場合はclassNameのみを返す', () => {
      expect(getClassNames('text')).toEqual(['text'])
    })

    it('cautionが入る場合', () => {
      expect(getClassNames('text', true)).toEqual(['text', '-caution'])
    })

    it('smallが入る場合', () => {
      expect(getClassNames('text', false, true)).toEqual(['text', '-small'])
    })
  })
})
