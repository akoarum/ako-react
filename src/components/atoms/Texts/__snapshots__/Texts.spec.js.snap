// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texts initialize snapshot 1`] = `
<div
  className="sc-bdVaJa ewEUte"
/>
`;
